packages/location-field/src/index.js

```diff
@@ -0,0 +1,831 @@
+   * must be referencable in the stopsIndex prop.
+   */
+  nearbyStops: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * Invoked whenever the text input is clicked or when the clear button is
+   * clicked.
+   */
+  onTextInputClick: PropTypes.func,
+  /**
+   * A function to handle when a location is selected. This is always dispatched
+   * with an object of the following form:
+   *
+   * ```js
+   * {
+   *  locationType: string,
+   *  location: object
+   * }
+   * '''
+   */
+  onLocationSelected: PropTypes.func.isRequired,
+  /**
+   * A list of recent searches to show to the user
+   */
+  sessionSearches: PropTypes.arrayOf(
+    PropTypes.shape({
+      lat: PropTypes.number,
+      lon: PropTypes.number,
+      name: PropTypes.string
+    })
+  ),
+  /**
+   * Whether or not to show the clear button
+   */
+  showClearButton: PropTypes.bool,
+  /**
+   * Whether or not to show user settings dialog
+   */
+  showUserSettings: PropTypes.bool,
+  /**
+   * show autocomplete options as fixed/inline element rather than dropdown
+   */
+  static: PropTypes.bool,
+  /**
+   * An index of stops by StopId
+   */
+  stopsIndex: PropTypes.objectOf(
+    PropTypes.shape({
+      /**
+       * The stop code if the stop has one
+       */
+      code: PropTypes.string,
+      /**
+       * The distance from the user to the stop in meters
+       */
+      dist: PropTypes.number,
+      lat: PropTypes.number,
+      lon: PropTypes.number,
+      name: PropTypes.string,
+      routes: PropTypes.arrayOf(
+        PropTypes.shape({
+          longName: PropTypes.string,
+          shortName: PropTypes.string
+        })
+      )
+    })
+  ),
+  /**
+   * If true, do not show nearbyStops or current location as options
+   */
+  suppressNearby: PropTypes.bool,
+  /**
+   * An array of recent locations and places a user has searched for.
+   */
+  userLocationsAndRecentPlaces: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string,
+      /**
+       * Can be either 'home', 'work', or null
+       */
+      icon: PropTypes.string,
+      lat: PropTypes.number.isRequired,
+      lon: PropTypes.number.isRequired,
+      name: PropTypes.string.isRequired,
+      /**
+       * This represents the last time that this location was selected in a
+       * search
+       */
+      timestamp: PropTypes.number.isRequired,
+      /**
+       * One of: 'home', 'work', 'stop' or 'recent'
+       */
+      type: PropTypes.string.isRequired
+    })
+  )
+};
+
+LocationField.defaultProps = {
+  addLocationSearch: () => {},
+  clearLocation: () => {},
+  currentPosition: null,
+  findNearbyStops: () => {},
+  hideExistingValue: false,
+  inputPlaceholder: null,
+  location: null,
+  nearbyStops: [],
+  onTextInputClick: null,
+  sessionSearches: [],
+  showClearButton: true,
+  showUserSettings: false,
+  static: false,
+  stopsIndex: null,
+  suppressNearby: false,
+  userLocationsAndRecentPlaces: []
```

Comment:
Perhaps we should remove some of these `defaultProps` values. Someone may try to use LocationField expecting it to work out of the box with no props defined for certain actions and then spend time trying to figure out which props they need to provide. Know what I mean?

Reply:
I kinda think it's better to turn off everything by default. If users want to be more advanced they can explore the other options as they see fit.